sysgen/sysgen.go

```diff
@@ -288,10 +288,13 @@ func generateArg(name, typ string, a []string, structs map[string]Struct, unname
 		}
 		fmt.Fprintf(out, "PtrType{%v, Dir: %v, Type: StrConstType{%v, Val: \"%v\"}}", common(), fmtDir("in"), common(), a[0]+"\\x00")
 	case "int8", "int16", "int32", "int64", "intptr":
-		if want := 0; len(a) != want {
-			failf("wrong number of arguments for %v arg %v, want %v, got %v", typ, name, want, len(a))
+		if len(a) == 2 {
```

Comment:
> Yes. the form foo int32[0, 5-10, 50, 100-200] is what we want.

You don't need to support the full form right now (if you need just the range now). If we change the range format to "[begin-end]", we can always extend this to the full form later.

> I cannot open the 

Sorry. Sure. The code is:
```go
package main

import (
	"fmt"
)

func main() {
	s := "10-15"
	var lo, hi int64
	if _, err := fmt.Sscanf(s, "%d-%d", &lo, &hi); err != nil {
		failf("failed to parse int range: %v (%v)", s, err)
	}
	if lo >= hi {
		failf("bad int range: %v", s)
	}
	fmt.Println(lo, hi)
}
```